src/internal_events/codecs.rs

```diff
@@ -99,7 +101,35 @@ impl<'a> InternalEvent for EncoderSerializeError<'a> {
         emit!(ComponentEventsDropped {
             count: 1,
             intentional: false,
-            reason: "Failed serializing frame.",
+            reason,
+        });
+    }
+}
+
+#[derive(Debug)]
+pub struct EncoderWriteAllError<'a, E> {
+    pub error: &'a E,
+}
+
+impl<E: std::fmt::Display> InternalEvent for EncoderWriteAllError<'_, E> {
```

Comment:
Maybe just `WriteError`? WriteAll seems overly specific, would we have a different error for events that just Write some?

Reply:
Good point. To make it more generic I'll also make `count` a variable.

Reply:
Addressed in https://github.com/vectordotdev/vector/pull/14225/commits/de926d09da37f5ec27f2f52ddc4e163e8336944a